app/assets/src/components/views/discovery/DiscoveryView.jsx

```diff
@@ -1012,22 +1038,121 @@ class DiscoveryView extends React.Component {
     }
   };
 
+  handleEmptyStateModalClose = () => {
+    this.setState({
+      emptyStateModalOpen: false,
+    });
+  };
+
+  renderNoDataBanners = () => {
+    const { currentTab, emptyStateModalOpen, userDataCounts } = this.state;
+    const { visualizations } = this;
+
+    if (!userDataCounts) return null;
+
+    if (emptyStateModalOpen) {
+      localStorage.setItem("DiscoveryViewSeenBefore", "1");
+    }
+
+    switch (currentTab) {
+      case "projects":
+        if (userDataCounts.projectCount === 0) {
+          return emptyStateModalOpen ? (
+            <EmptyStateModal onClose={this.handleEmptyStateModalClose} />
```

Comment:
I'd prefer `EmptyStateModal` be named `WelcomeFirstTimeUserModal` to be more clear about when it shows up but not a big deal.

Reply:
Makes sense. The best to capture it right now would be both... because it only shows up if the user has no data (could have been invited to a project).
@happyimadesignr and @oliviabholmes, we should probably show it anyway to all first time users, no? Maybe we can adapt the message to include users that start the adventure in IDseq with data already. Wdyt?

Reply:
@tfrcarvalho We should show it to all first-time users. I can't think of a way to word it for both cases other than having 2 different messages. I think if we show the current message to first time users even if they have a project, I think the message still makes enough sense.

With new naming, would it be ModalFirstTimeUser?

Reply:
Indeed. Renamed.